tutorials/dataframe/df104_HiggsToTwoPhotons.py

```diff
@@ -0,0 +1,221 @@
+            "m_yy", "weight")
+
+# Run the event loop and create the plot
+
+# Set styles
+ROOT.gROOT.SetStyle("ATLAS")
+
+# Create canvas with pads for main plot and data/MC ratio
+c = ROOT.TCanvas("c", "", 700, 750)
+
+upper_pad = ROOT.TPad("upper_pad", "", 0, 0.35, 1, 1)
+lower_pad = ROOT.TPad("lower_pad", "", 0, 0, 1, 0.35)
+for p in [upper_pad, lower_pad]:
+    p.SetLeftMargin(0.14)
+    p.SetRightMargin(0.05)
+    p.SetTickx(False)
+    p.SetTicky(False)
+upper_pad.SetBottomMargin(0)
+lower_pad.SetTopMargin(0)
+lower_pad.SetBottomMargin(0.3)
+
+upper_pad.Draw()
+lower_pad.Draw()
+
+# Run the event loop
+ggh = hists["ggH"].GetValue()
+vbf = hists["VBF"].GetValue()
+data = hists["data"].GetValue()
+
+# Fit signal + background model to data
+upper_pad.cd()
+fit = ROOT.TF1("fit", "([0]+[1]*x+[2]*x^2+[3]*x^3)+[4]*exp(-0.5*((x-[5])/[6])^2)", 105, 160)
+fit.FixParameter(5, 125.0)
+fit.FixParameter(4, 119.1)
+fit.FixParameter(6, 2.39)
+fit.SetLineColor(2)
+fit.SetLineStyle(1)
+fit.SetLineWidth(2)
+data.Fit("fit", "", "E SAME", 105, 160)
+fit.Draw("SAME")
+
+# Draw background
+bkg = ROOT.TF1("bkg", "([0]+[1]*x+[2]*x^2+[3]*x^3)", 105, 160)
+for i in range(4):
+    bkg.SetParameter(i, fit.GetParameter(i))
+bkg.SetLineColor(4)
+bkg.SetLineStyle(2)
+bkg.SetLineWidth(2)
+bkg.Draw("SAME")
+
+# Draw data
+data.SetMarkerStyle(20)
+data.SetMarkerSize(1.2)
+data.SetLineWidth(2)
+data.SetLineColor(ROOT.kBlack)
+data.Draw("E SAME")
+data.SetMinimum(1e-3)
+data.SetMaximum(8e3)
+data.GetYaxis().SetLabelSize(0.045)
+data.GetYaxis().SetTitleSize(0.05)
+data.SetStats(0)
+data.SetTitle("")
+
+# Scale simulated events with luminosity * cross-section / sum of weights
+# and merge to single Higgs signal
+lumi = 10064.0
+ggh.Scale(lumi * 0.102 / 55922617.6297)
+vbf.Scale(lumi * 0.008518764 / 3441426.13711)
+higgs = ggh.Clone()
+higgs.Add(vbf)
+higgs.Draw("HIST SAME")
+
+# Draw ratio
+lower_pad.cd()
+
+line = ROOT.TLine(105, 0, 160, 0)
+line.SetLineColor(4)
+line.SetLineWidth(2)
+line.SetLineStyle(2)
+line.Draw("E")
+
+ratiofit = ROOT.TH1F("ratiofit", "ratiofit", 5500, 105, 160)
+ratiofit.Eval(fit)
+ratiofit.SetLineColor(2)
+ratiofit.SetLineStyle(1)
+ratiofit.SetLineWidth(2)
+ratiofit.Add(bkg, -1)
+ratiofit.SetMinimum(-125)
+ratiofit.SetMaximum(250)
+ratiofit.GetXaxis().SetLabelSize(0.08)
+ratiofit.GetXaxis().SetTitleSize(0.12)
+ratiofit.GetXaxis().SetTitleOffset(1.0)
+ratiofit.GetYaxis().SetLabelSize(0.08)
+ratiofit.GetYaxis().SetTitleSize(0.09)
+ratiofit.GetYaxis().SetTitle("Data - Bkg.")
+ratiofit.GetYaxis().CenterTitle()
+ratiofit.GetYaxis().SetTitleOffset(0.7)
+ratiofit.GetYaxis().SetNdivisions(503, False)
+ratiofit.SetTitle("")
+ratiofit.GetXaxis().SetTitle("m_{#gamma#gamma} [GeV]")
+ratiofit.Draw()
+
+ratio = data.Clone()
+ratio.Add(bkg, -1)
+ratio.Draw("E SAME")
+for i in range(1, data.GetNbinsX()):
+    ratio.SetBinError(i, data.GetBinError(i))
+
+# Hide last label of the upper pad
```

Comment:
@couet There is a better solution for this? I would prefer not to put such hacks into the tutorials.

Reply:
Regarding the hiding of the last label with a pad.

Reply:
Fixed with `ChangeLabel`